homeassistant/components/decora_wifi/light.py

```diff
@@ -87,7 +87,7 @@ class DecoraWifiLight(LightEntity):
     def __init__(self, switch):
         """Initialize the switch."""
         self._switch = switch
-        self._unique_id = f"{DOMAIN}_{switch}"
+        self._unique_id = f"{DOMAIN}_{self._switch.name}"
```

Comment:
Name cannot be used as a unique identifier, see:

https://developers.home-assistant.io/docs/entity_registry_index/#unique-id-requirements

Reply:
Thanks, I'll see what I can do

Reply:
I have one of these switches and have messed around with the python-decora_wifi a bit; the switch's attribute dictionary contains the device mac address and serial number. If I remember correctly, it should be possible to retrieve the device's MAC address with
`self._switch.mac`


Was looking into writing the patch myself but I'm pretty green as an open source contributor so I'm still learning the tools and testing process.  Thanks to both of you for the quick look!